subsys/net/lib/ocpp/ocpp_j.c

```diff
@@ -0,0 +1,883 @@
+{
+	int ret = 0;
+
+	ret = parse_idtag_info(json, idtag_info);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int parse_bootnotification_msg(char *json, struct boot_notif *binfo)
+{
+	int ret = 0;
+
+	struct json_obj_descr descr[] = {
+		JSON_OBJ_DESCR_PRIM_NAMED(struct json_bootnotif_payload, "status",
+					  status, JSON_TOK_STRING),
+		JSON_OBJ_DESCR_PRIM_NAMED(struct json_bootnotif_payload, "interval",
+					  interval, JSON_TOK_NUMBER),
+		JSON_OBJ_DESCR_PRIM_NAMED(struct json_bootnotif_payload, "currentTime",
+					  current_time, JSON_TOK_STRING),
+	};
+
+	struct json_bootnotif_payload msg = { 0 };
+
+	ret = json_obj_parse(json, strlen(json), descr, ARRAY_SIZE(descr), &msg);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (!msg.status) {
+		return -EINVAL;
+	}
+
+	switch (*msg.status) {
+	case 'A':	/* accepted */
+		binfo->status = BOOT_ACCEPTED;
+		break;
+	case 'P':	/* pending */
+		binfo->status = BOOT_PENDING;
+		break;
+	case 'R':	/* rejected */
+		binfo->status = BOOT_REJECTED;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	if (!msg.interval) {
+		return -EINVAL;
+	}
+
+	binfo->interval = msg.interval;
+
+	if (!msg.current_time) {
+		return -EINVAL;
+	}
+
+	/* todo: convert civil time to epoch and update local time */
+	(void)binfo->date;
+
+	return 0;
+}
+
+static int parse_start_txn_msg(char *json,
+			       int *idtxn,
+			       struct ocpp_idtag_info *idtag_info)
+{
+	int ret = 0;
+
+	struct json_obj_descr descr[] = {
+		JSON_OBJ_DESCR_PRIM_NAMED(struct json_common_payload_field, "transactionId",
+					  val1, JSON_TOK_NUMBER),
+	};
+
+	struct json_common_payload_field payload = { 0 };
+
+	ret = json_obj_parse(json, strlen(json), descr, ARRAY_SIZE(descr), &payload);
+	if (ret < 0) {
+		return ret;
+	}
+
+	*idtxn = payload.val1;
+
+	ret = parse_idtag_info(json, idtag_info);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int parse_getconfig_msg(char *json, char *key)
+{
+	int ret = 0;
+
+	struct json_obj_descr descr[] = {
+		JSON_OBJ_DESCR_ARRAY_NAMED(struct json_getconfig_payload, "key",
+					   key, 1, key_len, JSON_TOK_STRING),
+	};
+
+	struct json_getconfig_payload payload = { 0 };
+
+	ret = json_obj_parse(json, strlen(json), descr,
+			     ARRAY_SIZE(descr), &payload);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* key is optional so return success*/
+	if (payload.key[0] != NULL) {
+		strcpy(key, payload.key[0]);
+	}
+
+	return 0;
+}
+
+static int parse_changeconfig_msg(char *json, char *key, char *val)
+{
+	int ret = 0;
+
+	struct json_obj_descr descr[] = {
+		JSON_OBJ_DESCR_PRIM_NAMED(struct json_common_payload_field_str, "key",
+					  val1, JSON_TOK_STRING),
+		JSON_OBJ_DESCR_PRIM_NAMED(struct json_common_payload_field_str, "value",
+					  val2, JSON_TOK_STRING),
+	};
+
+	struct json_common_payload_field_str payload = { 0 };
+
+	ret = json_obj_parse(json, strlen(json), descr, ARRAY_SIZE(descr), &payload);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (!payload.val1 || !payload.val2) {
```

Comment:
The code is full of these kind of checks and we are trying to get rid of them because of some MISRA rule. As this is new code we should avoid introducing code that we need to change later anyway.
Could you try to change this here (and also in other parts of the code) to explicit checks like this here 
```
if (payload.val1 == 0 || payload.val2 == 0) {`
```